identity/aziot-identity-client-async/src/lib.rs

```diff
@@ -4,37 +4,218 @@
+        let body = get_device_identity::Request {
+            id_type: ID_TYPE_AZIOT.to_string(),
+        };
+
+        let res: get_device_identity::Response = http_common::request(
+            &self.inner,
+            http::Method::POST,
+            make_uri!("/identities/device", self.api_version),
+            Some(&body),
+        )
+        .await?;
+
+        Ok(res.identity)
+    }
+
     pub async fn reprovision(&self) -> Result<(), std::io::Error> {
-        let body = aziot_identity_common_http::reprovision_device::Request {
-            id_type: "aziot".to_owned(),
+        let body = reprovision_device::Request {
+            id_type: ID_TYPE_AZIOT.to_string(),
         };
 
         http_common::request_no_content(
             &self.inner,
             http::Method::POST,
-            &format!(
-                "http://identityd.sock/identities/device/reprovision?api-version={}",
-                self.api_version
-            ),
+            make_uri!("/device/reprovision", self.api_version),
+            Some(&body),
+        )
+        .await?;
+
+        Ok(())
+    }
+
+    pub async fn create_module_identity(
+        &self,
+        module_name: &str,
+    ) -> Result<Identity, std::io::Error> {
+        let body = create_module_identity::Request {
+            id_type: ID_TYPE_AZIOT.to_string(),
+            module_id: module_name.to_string(),
+            opts: None,
+        };
+
+        let res: create_module_identity::Response = http_common::request(
+            &self.inner,
+            http::Method::POST,
+            make_uri!("/identities/modules", self.api_version),
+            Some(&body),
+        )
+        .await?;
+
+        Ok(res.identity)
+    }
+
+    pub async fn create_local_identity(
+        &self,
+        module_name: &str,
+        opts: Option<aziot_identity_common::LocalIdOpts>,
+    ) -> Result<Identity, std::io::Error> {
+        #[allow(clippy::redundant_closure)] // closure needed for map()
+        let body = create_module_identity::Request {
+            id_type: ID_TYPE_LOCAL.to_string(),
+            module_id: module_name.to_string(),
+            opts: opts.map(|opts| create_module_identity::CreateModuleOpts::LocalIdOpts(opts)),
+        };
+
+        let res: create_module_identity::Response = http_common::request(
+            &self.inner,
+            http::Method::POST,
+            make_uri!("/identities/modules", self.api_version),
             Some(&body),
         )
         .await?;
 
+        Ok(res.identity)
+    }
+
+    pub async fn update_module_identity(
+        &self,
+        module_name: &str,
+    ) -> Result<Identity, std::io::Error> {
+        let res: update_module_identity::Response = http_common::request::<(), _>(
+            &self.inner,
+            http::Method::PUT,
+            make_uri!(
+                "/identities/modules",
+                self.api_version,
+                ID_TYPE_AZIOT.to_string(),
```

Comment:
This `.to_string()` isn't needed. Same for the next three fns.

Reply:
Fixed